File: backend/job-service/eureka_client.py
import os
from py_eureka_client import eureka_client
from dotenv import load_dotenv

load_dotenv(os.path.join(os.path.dirname(__file__), '../.env'), override=True)

def register_eureka():
    e_url = os.getenv("EUREKA_URL", "http://localhost:5001")
    s_name = "JOB-SERVICE"
    s_port = int(os.getenv("JOB_SERVICE_PORT", 5004))
    hostname = os.getenv("HOSTNAME", "localhost")
    
    try:
        eureka_client.init(
            eureka_server=f"{e_url}/eureka",
            app_name=s_name,
            instance_port=s_port,
            instance_host=hostname,
            instance_ip="127.0.0.1",
            should_discover=False,
            renewal_interval_in_secs=30,
            duration_in_secs=90
        )
        print(f"Registered: {s_name} -> 127.0.0.1:{s_port}")
    except Exception as e:
        print(f"Eureka registration failed: {e}")

File: backend/user-service/eureka-client.js
const Eureka = require('eureka-js-client').Eureka;
const path = require('path');
require('dotenv').config({ path: path.resolve(__dirname, '../.env') });

const port = parseInt(process.env.USER_SERVICE_PORT, 10) || 5005;

const client = new Eureka({
  instance: {
    app: 'USER-SERVICE',
    instanceId: `USER-SERVICE:${port}`,
    hostName: process.env.HOSTNAME || 'localhost',
    ipAddr: '127.0.0.1',
    statusPageUrl: `http://127.0.0.1:${port}/user`,
    healthCheckUrl: `http://127.0.0.1:${port}/user`,
    port: {
      '$': port,
      '@enabled': 'true',
    },
    vipAddress: 'USER-SERVICE',
    dataCenterInfo: {
      '@class': 'com.netflix.appinfo.InstanceInfo$DefaultDataCenterInfo',
      name: 'MyOwn',
    },
    leaseInfo: {
      renewalIntervalInSecs: 30,
      durationInSecs: 90,
    },
  },
  eureka: {
    host: 'localhost',
    port: process.env.EUREKA_PORT || 5001,
    servicePath: '/eureka/apps/',
    maxRetries: 10,
    requestRetryDelay: 2000,
    heartbeatInterval: 30000,
    registryFetchInterval: 30000,
  },
});

function startEureka() {
  client.logger.level('warn');
  client.start(error => {
    if (error) {
      console.error('Eureka registration failed:', error);
    } else {
      console.log(`Registered: USER-SERVICE -> 127.0.0.1:${port}`);
    }
  });

  // Handle heartbeat errors
  client.on('heartbeat', () => {
    // Heartbeat successful
  });

  client.on('registryUpdated', () => {
    // Registry updated
  });
}

module.exports = { startEureka, eurekaClient: client };
